src/index.js

```diff
@@ -5,59 +5,63 @@ export default function ({ locals = {}, filters = {}, useMetadata = false } = {}
   const opts = arguments[0] || {}
 
   return (files, metalsmith, done) => {
-    setImmediate(done)
-
     // extend with metalsmith global metadata
     if (useMetadata) {
       locals = Object.assign({}, locals, metalsmith.metadata())
     }
 
-    Object.keys(files).forEach((file) => {
-      if (!/\.(pug|jade)/.test(path.extname(file))) {
-        return
-      }
+    try {
```

Comment:
Well, I prefer to bail-out on first error, but I can quickly change it to try all files.